prototype1/src/App.tsx

```diff
@@ -0,0 +1,125 @@
+import { useEffect, useRef } from "react"
+import "./App.css"
+import { data } from "./data"
+import * as d3 from "d3"
+import { useWindowSize } from "react-use"
+
+let { children } = data
+let id = (x: any) => x
+
+function useMount(fn: Function) {
+  useEffect(() => fn(), [])
+}
+
+const padding = 30
+const textSpacingFromCircle = 5
+
+function App() {
+  let svgRef = useRef<SVGSVGElement>(null)
+  let rootRef =
+    useRef<d3.Selection<SVGGElement, unknown, null, undefined>>(null)
+  let sizeProps = useWindowSize(0, 0)
+  let paddedSizeProps = {
+    height: sizeProps.height - padding * 2,
+    width: sizeProps.width - padding * 2,
+  }
+
+  useEffect(() => {
```

Comment:
Can the contents of this file maybe be split into some functions so that it is easier to read?